src/util/auth.rs

```diff
@@ -2,17 +2,45 @@ use argon2::{password_hash::PasswordHashString, Argon2, PasswordVerifier};
 use axum::{
     headers::{authorization::Bearer, Authorization},
     http::StatusCode,
+    TypedHeader,
 };
 
-/// Checks if user is authorized by checking if the given Bearer Token matches the given hash  
-/// Returns 401 (UNAUTHORIZED) with appropriate message if they do not match
-pub fn check_api_key(
+pub fn check_auth(
+    auth_query: Option<&String>,
+    auth_header: Option<TypedHeader<Authorization<Bearer>>>,
+    hash: &PasswordHashString,
+) -> Result<(), (StatusCode, String)> {
+    if let Some(auth) = auth_query {
+        return check_auth_query(auth, hash);
+    }
+
+    if let Some(TypedHeader(auth)) = auth_header {
+        return check_auth_header(auth, hash);
+    }
+
+    Err((StatusCode::UNAUTHORIZED, "Authorization failed!".to_owned()))
+}
+
+/// Checks if user is authorized by checking if the given Bearer Token matches the given hash
+pub fn check_auth_header(
     authorization: Authorization<Bearer>,
     hash: &PasswordHashString,
 ) -> Result<(), (StatusCode, String)> {
-    return match (Argon2::default())
-        .verify_password(authorization.token().as_bytes(), &hash.password_hash())
-    {
+    return check_auth_key(authorization.token().as_bytes(), hash);
+}
+
+/// Checks if user is authorized by checking if a given query parameter matches the given hash
+pub fn check_auth_query(
+    authorization: &String,
+    hash: &PasswordHashString,
+) -> Result<(), (StatusCode, String)> {
+    return check_auth_key(authorization.as_bytes(), hash);
+}
```

Comment:
Ist diese Funktion wirklich notwenidg? Das kann man doch inline in `check_auth` machen, oder nicht?

Reply:
Dachte mir, falls man in Zukunft nur über Query auth anbieten will wäre es sinnvoll. So ist halt Header/Query auth schön getrennt, kann es aber auch inlinen

Reply:
Würde es dann (wenn bzw. falls es dazu kommt) erst umbauen und hier der Einfachheit inlinen. Bin aber auch fine mit extra Methode, nur aktuell macht die halt basically nix.